app/queries/servers/categories.ts

```diff
@@ -35,6 +36,46 @@ export const queryCategoriesByTeamIds = (database: Database, teamIds: string[])
     return database.get<CategoryModel>(CATEGORY).query(Q.where('team_id', Q.oneOf(teamIds)));
 };
 
+export async function prepareCategoriesAndCategoriesChannels(operator: ServerDataOperator, categories: CategoryWithChannels[], prune = false) {
+    try {
+        const modelPromises: Array<Promise<Model[]>> = [];
+        const preparedCategories = prepareCategories(operator, categories);
+        if (preparedCategories) {
+            modelPromises.push(preparedCategories);
+        }
+
+        const preparedCategoryChannels = prepareCategoryChannels(operator, categories);
+        if (preparedCategoryChannels) {
+            modelPromises.push(preparedCategoryChannels);
+        }
+
+        const models = await Promise.all(modelPromises);
+        const flattenedModels = models.flat();
+
+        if (prune && categories.length) {
+            const remoteCategoryIds = new Set(categories.map((cat) => cat.id));
+
+            // If the passed categories have more than one team, we want to update across teams
+            const teamIds = pluckUnique('team_id')(categories) as string[];
+            const localCategories = await queryCategoriesByTeamIds(operator.database, teamIds).fetch();
+            const customCategories = localCategories.filter((c) => c.type === 'custom');
+            for await (const custom of customCategories) {
+                if (!remoteCategoryIds.has(custom.id)) {
+                    const categoryChannels = await custom.categoryChannels.fetch();
+                    for (const cc of categoryChannels) {
```

Comment:
Is the fix about deleting both the CategoryChannel and the Category?

Reply:
@avinashlng1080 yes, previously it was only deleting the category and left the orphan categoryChannels data in the database. By fixing the reported bug now I'm also fixing the removal of the related data to not have orphan data in the db